File: src/main/java/practice/account/application/in/CreateAccountUseCase.java
package practice.account.application.in;

import practice.account.domain.Account;
import practice.account.domain.ExternalAccount;

public interface CreateAccountUseCase {

  Account createMainAccount(Long userId);
  Account createSavingAccount(Long userId);
  ExternalAccount registerExternalAccount(Long userId, String bankCode, String accountNumber);

}

File: src/main/kotlin/practice/account/application/in/CreateAccountUseCase.kt
package practice.account.application.`in`

import practice.account.domain.Account
import practice.account.domain.ExternalAccount

interface CreateAccountUseCase {

    fun createMainAccount(userId: Long): Account
    fun createSavingAccount(userId: Long): Account
    fun registerExternalAccount(userId: Long, bankCode: String, accountNumber: String): ExternalAccount

}